ostd/src/task/task.rs

```diff
@@ -42,66 +45,42 @@ pub trait TaskContextApi {
 }
 
 pub struct KernelStack {
-    segment: Segment,
+    kva: Kva,
+    end_vaddr: Vaddr,
     has_guard_page: bool,
 }
 
 impl KernelStack {
     pub fn new() -> Result<Self> {
+        let mut new_kva = Kva::new(KERNEL_STACK_SIZE);
+        let pages = allocator::alloc::<KernelStackMeta>(KERNEL_STACK_SIZE).unwrap();
```

Comment:
It seems that in this way the stack pages will not be initialized to be zeroed, maybe add some `fill_zero()` instructions?. I think a new stack pages are better backed by fresh physical pages, even though this operation will introduce some overhead.